yuyo/repeaters.py

```diff
@@ -0,0 +1,228 @@
+        self._after = coro
+
+
+def with_ignored_exceptions(*exceptions: type) -> Callable[[Repeater[RepeaterFuncT]], Repeater[RepeaterFuncT]]:
+    """
+    Set the exceptions that a task will ignore.
+
+    If any of these exceptions are encountered, there will be nothing printed to console
+
+    Parameters
+    ----------
+    exceptions
+        List of exception types
+
+    Examples
+    --------
+    ```py
+    @yuyo.with_ignored_exceptions(ZeroDivisionError)
+    @yuyo.as_repeater(seconds=1)
+    async def repeater():
+        global run_count
+        run_count += 1
+        print(f"Run #{run_count}")
+    ```
+    """
+    for exception in exceptions:
+        if not issubclass(exception, Exception):
+            raise TypeError(f"Ignored exception must derive from Exception, is {exception.__name__}")
+
+    def decorator(repeater: Repeater[RepeaterFuncT]) -> Repeater[RepeaterFuncT]:
+        repeater.ignored_exceptions = exceptions
+        return repeater
+
+    return decorator
+
+
+def with_fatal_exceptions(*exceptions: type) -> Callable[[Repeater[RepeaterFuncT]], Repeater[RepeaterFuncT]]:
+    """
+    Set the exceptions that will stop a task.
+
+    If any of these exceptions are encountered, the task will stop.
+
+    Parameters
+    ----------
+    exceptions
+        List of exception types
+
+    Examples
+    --------
+    ```py
+    @yuyo.with_fatal_exceptions(ZeroDivisionError, RuntimeError)
+    @yuyo.as_repeater(seconds=1)
+    async def repeater():
+        global run_count
+        run_count += 1
+        print(f"Run #{run_count}")
+    ```
+    """
+    for exception in exceptions:
+        if not issubclass(exception, Exception):
+            raise TypeError(f"Fatal exception must derive from Exception, is {exception.__name__}")
+
+    def decorator(repeater: Repeater[RepeaterFuncT]) -> Repeater[RepeaterFuncT]:
+        repeater.fatal_exceptions = exceptions
+        return repeater
+
+    return decorator
+
+
+def as_repeater(
+    *,
+    seconds: Optional[float] = None,
+    minutes: Optional[float] = None,
+    hours: Optional[float] = None,
+    run_count: Optional[int] = None,
+    loop: Optional[asyncio.AbstractEventLoop] = None,
+) -> Callable[[RepeaterFuncT], Repeater[RepeaterFuncT]]:
+    """
+    Register a :class:Repeater.
+
+    Parameters
+    ----------
+    seconds : typing.Optional[float]
+        The number of seconds between repeater iterations
+    minutes : typing.Optional[float]
+        The number of minutes between repeater iterations
+    hours : typing.Optional[float]
+        The number of hours between repeater iterations
+    run_count : typing.Optional[int]
+        The number of times the repeater should run
+    loop : typing.Optional[asyncio.AbstractEventLoop]:
+        The event loop to use. If this is unspecified, the repeater defaults to :func:`asyncio.get_event_loop`
+    """
+
+    def decorator(function: RepeaterFuncT) -> Repeater[RepeaterFuncT]:
```

Comment:
We can and do make arguments positional only quite commonly since this is 3.8+ (so using `, /,`) so that should be used for cases like this and the `__init__`. Also "callback" not function.